File: java_solutions/628.maximum-product-of-three-numbers.java
/*
 * @lc app=leetcode id=628 lang=java
 *
 * [628] Maximum Product of Three Numbers
 */

class Solution {
    public int maximumProduct(int[] nums) {
        return Optional.of(Arrays.stream(nums).sorted().toArray()).map(s -> Math.max(s[s.length - 1] * s[s.length - 2] * s[s.length - 3], s[s.length - 1] * s[0] * s[1])).get();
    }
}

File: kotlin_solutions/628.maximum-product-of-three-numbers.kt
/*
 * @lc app=leetcode id=628 lang=kotlin
 *
 * [628] Maximum Product of Three Numbers
 */

class Solution { fun maximumProduct(nums: IntArray): Int = nums.sorted().let { s -> maxOf(s[s.lastIndex] * s[s.lastIndex - 1] * s[s.lastIndex - 2], s[0] * s[1] * s[s.lastIndex]) } }
